tests/internal/aws_s3_local_buffer.c

```diff
@@ -0,0 +1,246 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+#include <fluent-bit/flb_sds.h>
+#include <fluent-bit/flb_s3_local_buffer.h>
+#include <fluent-bit/flb_output_plugin.h>
+#include <fluent-bit/flb_mem.h>
+#include <fluent-bit/flb_info.h>
+#include <fluent-bit/flb_aws_util.h>
+
+#include "flb_tests_internal.h"
+
+#define BUFFER_DIRECTORY FLB_TESTS_DATA_PATH "data/s3_local_buffer/"
+#define TEST_DATA "I love Fluent Bit"
+#define S3_KEY_FORMAT_SPECIFIC_TAG "logs/$TAG[2]/$TAG[0]/%Y/%m/%d"
+#define S3_OBJECT_KEY_SPECIFIC_TAG "logs/ccc/aa/2020/08/15"
+#define S3_KEY_FORMAT_FULL_TAG "logs/$TAG/%Y/%m/%d"
+#define S3_OBJECT_KEY_FULL_TAG "logs/aa.bb.ccc/2020/08/15"
+#define S3_KEY_FORMAT_SPECIAL_CHARCATERS_TAG "logs/my.great_photos-2020:jan/$TAG/%Y/%m/%d"
+#define S3_OBJECT_KEY_SPECIAL_CHARCATERS_TAG "logs/my.great_photos-2020:jan/aa.bb.ccc/2020/08/15"
+#define TAG "aa.bb.ccc"
+#define MULTI_DELIMITER_TAG "aa.bb-cc"
+#define TAG_DELIMITER "."
+#define TAG_DELIMITERS ".,-"
+#define PLUGIN_NAME "s3_plugin"
```

Comment:
To make this easier to read, put all of the constants used in one test next to each other, and separate the ones for each test with a newline, so that its easy to see what all goes together